[BUG] Error linking against the civetweb libraries unzipped with zipfile Python module

<!-- :warning: Please, try to follow the template -->

### System information

- **RTI Product**:
- **Version**:
- **Operating system**:
- **Compiler**:
- **Compiler version**:
- **Additional information**:

### What is the current behavior?

When a zipped Connext installation is extracted using the `zipfile` Python module, the symbolic links are not taken into account. As a result we get the following error when we try to link against the civetweb libraries:

```
Scanning dependencies of target WebIntegrationServiceLibraryAPICXX11
[ 99%] Building CXX object web_integration_service/library_api/c++11/CMakeFiles/WebIntegrationServiceLibraryAPICXX11.dir/web_integration_service_library_api.cxx.o
[ 99%] Linking CXX executable WebIntegrationServiceLibraryAPICXX11
/usr/bin/ld:/rti/jenkins/workspace/ci_rticommunity-cmake-utils_main/rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb-cpp.so: file format not recognized; treating as linker script
/usr/bin/ld:/rti/jenkins/workspace/ci_rticommunity-cmake-utils_main/rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb-cpp.so:0: syntax error
collect2: error: ld returned 1 exit status
make[2]: *** [web_integration_service/library_api/c++11/CMakeFiles/WebIntegrationServiceLibraryAPICXX11.dir/build.make:94: web_integration_service/library_api/c++11/WebIntegrationServiceLibraryAPICXX11] Error 1
make[1]: *** [CMakeFiles/Makefile2:20752: web_integration_service/library_api/c++11/CMakeFiles/WebIntegrationServiceLibraryAPICXX11.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

File check:

```
$> ls -la rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/
total 600K
drwxr-xr-x 3 luis luis 4,0K jul  3 19:45 .
drwxr-xr-x 3 luis luis 4,0K jul  3 19:45 ..
...
-rwxrwxrwx 1 luis luis   25 jul  3 19:45 libcivetweb-cpp.so
-rw-r--r-- 1 luis luis  47K jul  3 19:45 libcivetweb-cpp.so.1.15.0
-rwxrwxrwx 1 luis luis   21 jul  3 19:45 libcivetweb.so
-rw-r--r-- 1 luis luis 203K jul  3 19:45 libcivetweb.so.1.15.0

$ file rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb.so 
rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb.so: ASCII text, with no line terminators

$ cat rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb.so 
libcivetweb.so.1.15.0
```

### Steps to reproduce the issue

### Expected behavior

```
$ ls -la rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/
total 592K
drwxr-xr-x 3 luis luis 4,0K jun 11 00:30 .
drwxr-xr-x 3 luis luis 4,0K jun 11 00:30 ..
...
lrwxrwxrwx 1 luis luis   25 jul  3 19:49 libcivetweb-cpp.so -> libcivetweb-cpp.so.1.15.0
-rw-r--r-- 1 luis luis  47K jun 10 07:34 libcivetweb-cpp.so.1.15.0
lrwxrwxrwx 1 luis luis   21 jul  3 19:49 libcivetweb.so -> libcivetweb.so.1.15.0
-rw-r--r-- 1 luis luis 203K jun 10 07:34 libcivetweb.so.1.15.0

$ file rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb.so 
rti_connext_dds-7.2.0/third_party/civetweb-1.15/x64Linux4gcc7.3.0/release/lib/libcivetweb.so: symbolic link to libcivetweb.so.1.15.0
```

### Suggested solutions

Instead of using the built-in module, use the command-line tool mode of CMake to extract the ZIP file.

--- a/resources/ci_cd/linux_install.py
+++ b/resources/ci_cd/linux_install.py
@@ -14,39 +14,20 @@ The environment variable RTI_MIN_PACKAGE_URL must be assigned for downloading
 the minimal installation archive.
 
 """
-import io
 import os
 import sys
-import zipfile
+import tempfile
+import urllib.request
 
+from distutils.spawn import find_executable
 from pathlib import Path
-from urllib import request
+from subprocess import call
 from urllib.error import URLError
-from zipfile import ZipFile, ZipInfo
-
-
-class ZipFileWithPermissions(ZipFile):
-    """Custom ZipFile class handling file permissions.
-
-    Code from https://stackoverflow.com/a/54748564
-    """
-
-    def _extract_member(self, member, targetpath, pwd):
-        if not isinstance(member, ZipInfo):
-            member = self.getinfo(member)
-
-        targetpath = super()._extract_member(member, str(targetpath), pwd)
-        attr = member.external_attr >> 16
-
-        if attr != 0:
-            Path(targetpath).chmod(attr)
-
-        return targetpath
 
 
 def main():
     rti_minimal_package_url = os.getenv("RTI_MIN_PACKAGE_URL")
-
+    temp_dir = Path(tempfile.gettempdir())
     if not rti_minimal_package_url:
         sys.exit(
             "Environment variable RTI_MIN_PACKAGE_URL not found, skipping..."
@@ -59,23 +40,38 @@ def main():
     except FileNotFoundError:
         sys.exit("The RTI_INSTALLATION_PATH does not exist.")
 
+    cmake_command = find_executable("cmake")
+    if cmake_command is None:
+        sys.exit("CMake must be installed in order to use the script.")
+
     try:
-        resp = request.urlopen(rti_minimal_package_url)
+        rti_zipped_file_name = Path(
+            urllib.request.url2pathname(rti_minimal_package_url)
+        ).name
+        rti_zipped_file_path = temp_dir.joinpath(rti_zipped_file_name)
+        urllib.request.urlretrieve(
+            rti_minimal_package_url, rti_zipped_file_path
+        )
     except URLError as e:
         sys.exit("Error opening the URL: {}".format(e))
 
-    print("Extracting minimal installation.")
-
     try:
-        with ZipFileWithPermissions(io.BytesIO(resp.read())) as rti_zipfile:
-            bad_file_name = rti_zipfile.testzip()
+        return_value = call(
+            [
+                cmake_command,
+                "-E",
+                "tar",
+                "xf",
+                str(rti_zipped_file_path),
+                "--format=zip",
+            ],
+            cwd=rti_installation_path,
+        )
+    except FileNotFoundError:
+        sys.exit("The CMake executable could not be found.")
 
-            if bad_file_name:
-                sys.exit("Bad file found in the archive.")
-
-            rti_zipfile.extractall(rti_installation_path)
-    except zipfile.BadZipFile as e:
-        sys.exit("Error opening zip file: {}".format(e))
+    if return_value:
+        sys.exit("There were error extracting the package")
 
 
 if __name__ == "__main__":

--- a/resources/ci_cd/linux_static_analysis.py
+++ b/resources/ci_cd/linux_static_analysis.py
@@ -52,7 +52,7 @@ def find_connext_dir() -> Path:
 def parse_args() -> argparse.Namespace:
     """Parse the CLI options."""
     parser = argparse.ArgumentParser()
-    parser.add_argument("--build-dir", type=Path, default="examples/build")
+    parser.add_argument("--build-dir", type=Path, default="build")
     parser.add_argument("--connext-dir", type=Path)
     args = parser.parse_args()
 

